### ISSUE TITLE
MissingSuperCall false positive when the annotated function is in the grandparent

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#9468]
## Expected Behavior
If I call the super class MissingSuperCall doesn't raise any issue

## Observed Behavior
If the annotated function is in the grandparent instead of the parent the rule raises an issue even when you call super.

## Steps to Reproduce
Unit test:
```kotlin
@Test
fun `super super methods has the annotation 2`() {
    val subject = MissingSuperCall(
        TestConfig("mustInvokeSuperAnnotations" to listOf("Ann"))
    )

    val code = """
        annotation class Ann
        open class Foo {
            @Ann
            open fun x() {}
        }
        open class Bar: Foo() {
            override fun x() {
                super.x()
                println("Bar")
            }
        }
        class Baz: Bar() {
            override fun x() {
                super.x()
                println("Baz")
            }
        }
    """.trimIndent()
    val actual = subject.lintWithContext(env, code)
    assertThat(actual).isEmpty()
}
```

## Context
This patter is really common on Android.

## Your Environment
* Version of detekt used: main
